documentation/src/physical_types/spins/symbolic.md

```diff
@@ -0,0 +1,55 @@
+# Symbolic parameters
+
+Symbolic parameters, or parametrisation of the classes, can be very useful for users creating Hamiltonians or operators with different coupling/on-site/... strengths.
+
+## Example
+
+For instance, if a user would like to define the following Hamiltonian:
+
+\\[
+    \hat{H} = J \sum_{i, j} \sigma_x^i \sigma_x^j + \alpha \sum_i \sigma_z^i
+\\]
+
+where the values for \\(J\\) range between [-1.0, 1.0] and the values for \\( \alpha \\) range between [0.2, 0.4], the user might try to define all these Hamiltonians one at a time. However, with struqture, the user can define the Hamiltonian once, passing symbolic parameters `"J"` and `"alpha"` as the coefficients, and substituting them for the correct values in situ, later. This is shown below.
+
+```python
+from struqture_py.spins import PauliHamiltonian
+import py_alqorithms
+
+# Defining the base Hamiltonian
+hamiltonian = PauliHamiltonian()
+number_spins = 4
+for i in range(number_spins):
+    for j in (i + 1, number_spins):
+        hamiltonian.add_operator_product(f"{i}X{j}X", "J")
+for i in range(number_spins):
+    hamiltonian.add_operator_product(f"{i}Z", "alpha")
+
+print(hamiltonian)
+
+# This Hamiltonian can then be used later in the following fashion:
+algorithm = py_alqorithms.QSWAPAlgorithm(1)
```

Comment:
Are we ok with referencing non-open-source packages in the Documentation? If so we should mention what it is and where it belongs.